libretro/libretro-core.c

```diff
@@ -2374,88 +2596,13 @@ void retro_run(void)
    microSecCounter += (1000000/(retro_get_region() == RETRO_REGION_NTSC ? C64_NTSC_RFSH_PER_SEC : C64_PAL_RFSH_PER_SEC));
 }
 
-#define M3U_FILE_EXT ".m3u"
-
 bool retro_load_game(const struct retro_game_info *info)
 {
    if (info)
    {
-	const char *full_path;
-	bool is_fliplist = false;
-
-	(void)info;
-
-	full_path = info->path;
-
-	// If it's a m3u file
-	if(strendswith(full_path, M3U_FILE_EXT))
-	{
-		// Parse the m3u file
-		dc_parse_m3u(dc, full_path);
-		is_fliplist = true;
-	}
-	else if (strendswith(full_path, ".vfl"))
-	{
-		// Parse the vfl file
-		dc_parse_vfl(dc, full_path);
-		is_fliplist = true;
-	}
-
-	if (is_fliplist)
-	{
-		// Some debugging
-		log_cb(RETRO_LOG_INFO, "m3u file parsed, %d file(s) found\n", dc->count);
-		for(unsigned i = 0; i < dc->count; i++)
-		{
-			log_cb(RETRO_LOG_INFO, "file %d: %s\n", i+1, dc->files[i]);
-		}	
-	}
-	else
-	{
-		// Add the file to disk control context
-		// Maybe, in a later version of retroarch, we could add disk on the fly (didn't find how to do this)
-		dc_add_file(dc, full_path);
-	}
-
-	// Init first disk
-	dc->index = 0;
-	dc->eject_state = false;
-	if (dc->count != 0)
-	{
-	    int unit = get_image_unit();
-	    if (unit == 1)
-		log_cb(RETRO_LOG_INFO, "Tape (%d) inserted into datasette: %s\n", dc->index+1, dc->files[dc->index]);
-	    else 
-		log_cb(RETRO_LOG_INFO, "Disk (%d) inserted into drive %d: %s\n", dc->index+1, unit, dc->files[dc->index]);
-	    strcpy(RPATH,dc->files[0]);
-	}
-	else
-	{
-	    log_cb(RETRO_LOG_WARN, "No images found in list file %s\n", full_path);
-	    RPATH[0]=0;
-	}
-   }
-   else
-   {
-      RPATH[0]=0;
+      process_cmdline(info->path);

```

Comment:
Where are you inserting the disk that was selected (if it was a single d64 file)? You cannot assume pre_main is always executed when retro_load_game is called. Pre_main() is only executed when the core is loaded. On static platforms, that happens only when the core is first started, but not again when Load Content is later selected when the core is already running.

This needs a lot of changes to work correctly again on static platforms.

Reply:
I see the your point, but I don't think it will require much work to address. It just have to behave differently when pre_main has already been called. On the other hand some things won't be possible - the cmd file assumes that you can pass full command line to the x64/xvic/..., which can't be done once pre_main is already called, so the best thing we can to is just read what we can from the cmd - autostart image and/or fliplist and ignore the rest.
I'll take care of that (as soon as I get that bloody wiiu to compile on travis).

Reply:
> I'll take care of that (as soon as I get that bloody wiiu to compile on travis).

Don't worry about, it always fails. The core build does not fail, but for some reason it also builds RA, which for some other reason does fail.

Reply:
> Don't worry about it, it always fails.

Well, not anymore - see https://github.com/libretro/RetroArch/pull/9496